src/client/Microsoft.Identity.Client/AuthenticationResultMetadata.cs

```diff
@@ -24,7 +24,7 @@ public AuthenticationResultMetadata(TokenSource tokenSource)
         /// <summary>
         /// The source of the token in the result.
         /// </summary>
-        public TokenSource TokenSource { get; }
+        public TokenSource TokenSource { get; internal set; }
```

Comment:
might as well make the setter public, as it will make the object testable.

Reply:
it is already testable by passing in TokenSource to the constructor

Reply:
Will be reverting this back and using token source as IDP for proactive token refresh only for client side telemetry through otel.